scripts/changelog/README.md

```diff
@@ -0,0 +1,68 @@
+# Auto-generate changelog for ksqlDB releases
+
+The ksqlDB repo uses conventional commits in order to assist with auto-generation of changelogs.
+
+## One-time setup
+
+Install the [tool](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-cli) we use to for generating changelogs:
+```
+npm install -g conventional-changelog-cli
+```
+
+## Generate changelog
+
+Assuming you've completed the one-time setup above, steps for generating a changelog are as follows.
+
+Suppose you're interested in generating a changelog of all changes in ksqlDB 0.8.0 that are new as of ksqlDB 0.7.0. Refer to these versions as `NEW_VERSION` and `OLD_VERSION`, respectively. The same process applies for other new/old ksqlDB version combinations as well (e.g., 0.8.1 and 0.8.0, or 0.8.0 and 0.7.1).
+
+These steps assume you're in the root of the `ksql` directory.
+
+### List commits on `NEW_VERSION`
+
+Start by listing all commits from the `NEW_VERSION`, which requires checking out the relevant branch/tag/commit. If the release tag exists:
+```
+git fetch --all --tags --prune
+git checkout tags/v0.8.0-ksqldb
+```
+If the tag hasn't been created yet but a branch has been cut (e.g., if a release is in progress):
+```
+git checkout 0.8.x-ksqldb
+git fetch upstream
+git merge upstream/0.8.x-ksqldb
+```
+If the branch hasn't been cut yet (which could also be the case for a release currently in progress):
+```
+git checkout <GIT_SHA>
+```
+
+Once on the relevant branch/tag/commit, fetch the commits and save them to a file:
+```
+conventional-changelog -p angular -i new_commits.txt -s -r 0
+```
+
+### List commits on `OLD_VERSION`
+
+Next, repeat the process to list commits from the `OLD_VERSION`. A tag for this release should already exist, so check out the tag:
+```
+git fetch --all --tags --prune
+git checkout tags/v0.7.0-ksqldb
+```
+and generate the list of commits:
+```
+conventional-changelog -p angular -i old_commits.txt -s -r 0
+```
+
+### Run the script
+
+Now you're ready to run the script that generates a changelog based on the difference between the two lists of commits from above. In other words, the changelog consists of new commits on `NEW_VERSION` not present on `OLD_VERSION`.
+
+Switch back to master to ensure you have the latest version of the script:
+```
+git checkout master
+```
+and run
+```
+python3 scripts/changelog/process_commit_files.py --release_version 0.8.0 --previous_version 0.7.0 --output_file CHANGELOG_additions.md
```

Comment:
I get this error with the script:
```
$ python3 scripts/changelog/process_commit_files.py --release_version 0.8.0 --previous_version 0.7.0 --output_file CHANGELOG_additions.md
Traceback (most recent call last):
  File "scripts/changelog/process_commit_files.py", line 330, in <module>
    all_commits = parse_changelog(args.new_commits_file)
  File "scripts/changelog/process_commit_files.py", line 59, in parse_changelog
    fork_name = _get_fork_name(line)
  File "scripts/changelog/process_commit_files.py", line 124, in _get_fork_name
    start_ind = header.index('https://github.com/')
ValueError: substring not found
```

I noticed the commits file do not include the fork name:
`# [](/compare/v0.8.0-ksqldb...v) (2020-05-08)`

I don't know why the conventional-changelog does not include the https://github.com. The version I have is 2.0.21.

Reply:
What the output of `git remote -v` for you?

Reply:
Also to double-check, which branch/tag were you on when you generated the commit file (using the `conventional-changelog` command)?

Reply:
```
$ git remote -v
sergio  git@github.com:spena/ksql (fetch)
sergio  git@github.com:spena/ksql (push)
upstream        git@github.com:confluentinc/ksql (fetch)
upstream        git@github.com:confluentinc/ksql (push)
```

For the branch/tag question. I tried different options with same results. For instance:
* git checkout tags/v0.8.0-ksqldb
* git checkout tags/v0.8.0-ksqldb -b test

The above two ways allow me to run the conventional-changelog, but the commit files still have the same header. I just tried both.